src/platform/datapath_raw_dummy.c

```diff
@@ -0,0 +1,252 @@
+/*++
```

Comment:
What is the purpose of this file?

Reply:
Looks like you are trying to get UWP build to work. Can you not use `#ifdef QUIC_UWP_BUILD` to achieve the same goal?

Reply:
You mean all datapath API in `_win.c` to have `ifdef` for UWP?

Reply:
Not all datapath API usage. I think you only need to use `#ifdef QUIC_UWP_BUILD` (sorry I wrote `ifdef` earlier) at a few places. The most obvious place is the raw datapath initialization. You might still need to put some more pieces of code behind the macro that do not compile under UWP like where we use IP MIB structs or some other functions.

Looking at where we currently use `QUIC_UWP_BUILD`, IIUC, it seems like some functions are exported in onecore.lib (which is what UWP build links against) but the function declaration is missing/not exposed in the UWP kit. (If you search for `#ifdef QUIC_UWP_BUILD`, you will know what I am talking about). I wonder if you can do the same for the failures you were seeing for UWP build.

Reply:
As far as I saw before, there are a lot of type not found error in case of uwp.
Adding ifdef for such type would varying many places and it end up with building successfully, but the functionality would be gone for uwp even though we don't use raw for uwp. I'm not sure such change makes sense.

Reply:
I took a quick look at the compile errors in the failed build. Is it just CxPlatGetInterfaceRssQueueCount and RawResolveRoute that cannot compile for UWP? If so, I think just we can just whack them with `ifdef` like this:

```
#ifdef QUIC_UWP_BUILD
_IRQL_requires_max_(PASSIVE_LEVEL)
QUIC_STATUS
RawResolveRoute(
    _In_ CXPLAT_SOCKET* Sock,
    _Inout_ CXPLAT_ROUTE* Route,
    _In_ uint8_t PathId,
    _In_ void* Context,
    _In_ CXPLAT_ROUTE_RESOLUTION_CALLBACK_HANDLER Callback
    )
{
    return QUIC_SUCCESS;
}
#else
// actual definition
#endif
```

If there are more instances, then maybe a dummy file is better but we need to rename it and make it clear it's for UWP.

Reply:
"it end up with building successfully, but the functionality would be gone for uwp even though we don't use raw for uwp."
I don't quite understand this sentence. Did you mean we would end up having unreachable code in the binary?

Reply:
I also would prefer and #ifdef approach instead of dummy file.

Reply:
- Not only function using MIB, but also `afxdp_helper.h` depending data (like XSK_RING which is in `XDP_RX_PACKET{ XDP_QUEUE{ XSK_RING } }`) get affected
- Not non unreachable code, but non functional code.

Reply:
<img width="1155" alt="image" src="https://github.com/microsoft/msquic/assets/5763034/0b8f65fe-b14c-480b-9ca9-3a1bbd4b4aa0">

Reply:
We can leave the dummy file.